tools/clang/unittests/HLSLTestLib/FileCheckerTest.cpp

```diff
@@ -256,7 +256,7 @@ class IncludeHandlerVFSOverlayForTest : public IDxcIncludeHandler {
   }
 };
 
-static IncludeHandlerVFSOverlayForTest *AllocVFSIncludeHandler(IUnknown *pUnkLibrary, const FileMap *pVFS) {
+static CComPtr<IncludeHandlerVFSOverlayForTest> AllocVFSIncludeHandler(IUnknown *pUnkLibrary, const FileMap *pVFS) {
```

Comment:
Another solution would have been to modify users of `AllocVFSIncludeHandler` to properly `Attach()` to the pointer, since that's how the `Alloc*` function should be treated.

Reply:
Would you prefer that?

Reply:
If I do that, then we shouldn't need the conversion constructor template anymore, though I wonder if we'll hit this again in the future, since it's perfectly valid to use CComPtr that way.

Reply:
I think the conversion constructor is fine to keep.  I don't feel too strongly about the approach here, since either way is correct, and it is a static function in test support code.